src/app/hardfork_test/src/internal/hardfork/fork_config.go

```diff
@@ -0,0 +1,75 @@
+package hardfork
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+// ForkConfigData holds the expected fork configuration data
+type ForkConfigData struct {
+	BlockchainLength int    `json:"blockchain_length"`
+	GlobalSlot       int    `json:"global_slot_since_genesis"`
+	StateHash        string `json:"state_hash"`
+	NextSeed         string `json:"next_seed"`
+	StakingSeed      string `json:"staking_seed"`
+}
+
+// ExtractForkConfig extracts the fork configuration from the network
+func (t *HardforkTest) ExtractForkConfig(port int, forkConfigPath string) ([]byte, error) {
+	for attempt := 1; attempt <= t.Config.ForkConfigMaxRetries; attempt++ {
```

Comment:
I think this retry mechanism is a bit problematic. When encountering some errors, no retry happens at all